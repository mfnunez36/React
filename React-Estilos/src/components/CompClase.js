import React from 'react';

/*
    Podemos importar estilos que seran utilizados automaticamente 
    pero podemos importar estilos como modulos utilizando la palabra 
    module en el nombre del archivo ej.: nombreArchivo.module.css

    Podemos acceder a los modulos de estilos como objeto 
    con nombres simples pero para nombres compuestos debemos utilizar 
    [] ej.: styles.nombresimple - styles['nombre-compuesto']


    NOTA: 
    Importando estilos de forma automatica corremos 
    el riesgo de pisar estilos que tengan el mismo nombre pero 
    importando estilos en forma de modulo estos tendran nombres 
    con identificador unico por lo que es imposible que se pisen con 
    otros estilos con mismo nombre 
*/

// import './CompClase.css';

import styles from './CompClase.module.css'


/*COMPONENTE DE CLASE*/
class CompClase extends React.Component {
    
    state = {
        estado: 0
    }

    cambiar = () => { 
        this.setState({ 
            estado: this.state.estado = 1 
        });
    }

    volver = () => { 
        this.setState({ 
            estado: this.state.estado = 0 
        });
    }

    render = () => {
        /* 
            Para los estilos en REACT debemos utilizar camelcase para 
            los nombres compuestos ej.:

            const styles = {
                marginBottom: '1em',
                marginRight: '600px',
                borderRadius: '0.5em',
            }

            Podemos evaluar dentro de llaves para agregar logica ej.:
            const clases = `compClase ${ item ? 'compClase-change' : '' }`
        */
            

        const item = this.state.estado > 0;
        
        //utilizando estilos con importacio automatica
        //const clases = item ? 'compClase-change compClase' : 'compClase';
        //const clases = `compClase ${ item ? 'compClase-change' : '' }`
        

        //utilizando estilos con importacion de modulo
        const clases = `${ styles.compClase } ${ item ? styles['compClase-change'] : '' }`

        return (
            <div>
                <h3>{ this.props.name }</h3>
                <p> 
                    {/* en JSX se puede comentar con barras multilineas dentro de llaves */}

                    {/*  
                        Para agregar nuestros estilos debemos utilizar 
                        un objeto que los contenga enviandolos a la propiedads style: 
                        <div style={ styles }></div> 
                    */}
                    


                    {/* 
                        Para agregar estilos de clase debe ser con className 
                    */}
                    <div className={ clases }> Estado : { this.state.estado } </div>
                    <button onClick={ 
                         this.cambiar
                    }>Cambiar</button>

                    <button onClick={ 
                         this.volver
                    }>Volver</button>

                    <hr/>
                    { this.props.descripcion }
                    { this.props.numeros }
                </p>
            </div> 
        );    
    }
}

export default CompClase;