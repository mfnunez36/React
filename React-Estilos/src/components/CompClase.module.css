.compClase {
    border: 1px solid black;
    margin-bottom: 1em;
    margin-right: 600px;
    border-radius: 0.5em;
    padding: 1em;
    background: #FFF;
    color: #000;
    transition: all 400ms ease-out;
}

.compClase-change {
    background: linear-gradient(45deg, black, #4a02f7);
    color: #FFF;
}
