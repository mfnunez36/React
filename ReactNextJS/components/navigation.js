import Link from "next/link";

const Navigation = () => {
  return (
    <nav className="navbar navbar-expand-lg bg-light">
        <div className="container-fluid">
            <Link href={'/'}>
                <a className="navbar-brand">NextJS Ejemplo</a>
            </Link>
            <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarNavAltMarkup"
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
            >
            <span className="navbar-toggler-icon" />
            </button>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <Link href={'/about'}>
                        <a className="nav-link">About</a>
                    </Link>
                    <Link href={'/services'}>
                        <a className="nav-link">Services</a>
                    </Link>
                </div>
            </div>
        </div>
    </nav>
  );
};

export default Navigation;
