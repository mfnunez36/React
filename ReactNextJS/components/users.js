import Router from "next/router";

const Users = ({users}) => {
    return (
        <>
            <h2>Users</h2>
            <ul className="list-group">
                {
                    users.map(user => (
                        <li 
                        className="list-group-item list-group-item-action" 
                        key={ user.id }
                        onClick={(e)=>{ Router.push('/users/[id]', `/users/${user.id}`) }}
                        >
                            <h5>{ user.name }</h5>
                            <label>Email :</label><p>{ user.email }</p>
                            <label>Web site :</label><p><a href={`www`+user.website}>{ user.website }</a></p>
                        </li>
                    )) 
                }
            </ul>
        </>
    )
}

export default Users;