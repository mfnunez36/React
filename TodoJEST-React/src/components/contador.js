import React from 'react'
import { useState } from 'react'

export const Contador = () => {
    const [clicks, setClicks] = useState(0)

    const increment = () => { setClicks( clicks + 1 ) }
    const decrement = () => { setClicks( clicks - 1 ) }
    
    return (
        <div>
            <h3>{clicks}</h3>
            <button onClick={increment}>increment</button>
            <button onClick={decrement}>decrement</button>
        </div>
    )
}
