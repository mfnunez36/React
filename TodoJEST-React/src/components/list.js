import React from 'react'

export const List = ({title, list}) => {
  return (
    <section>
        <h1>{title}</h1>
        {
            list.map(item => (
                <ul key={item.id}>
                    <li>{item.name}</li>
                </ul>
            ))
        }
    </section>
  )
}
