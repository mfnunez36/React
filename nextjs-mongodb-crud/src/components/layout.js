import Navbar from "./navbar"

export const Layout = ({children}) => {
    return (
      <>
        <Navbar/>
        { children}
      </>
    )
  }
  