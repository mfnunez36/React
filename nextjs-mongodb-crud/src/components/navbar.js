/* eslint-disable @next/next/no-img-element */
import Link from 'next/link'
import { useRouter } from 'next/router'

const Navbar = () => {
    const router = useRouter()
    
    return (
        <nav className="navbar navbar-dark bg-dark">
            <div className="container-fluid">
                <Link href='/'>
                    <img width='3%' height='3%' className="navbar-brand" src='/favicon.ico' alt='logo'/>
                </Link>
                
                <div className="navbar-text">
                    <button className='btn btn-primary' onClick={() => router.push("/tareas/crear") }>Crear Tarea</button>
                </div>
            </div>
        </nav>
    )
}

export default Navbar