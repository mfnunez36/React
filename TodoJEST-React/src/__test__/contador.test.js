import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import { Contador } from '../components/contador'

configure({
    adapter: new Adapter()
})

describe('Probando Componente Contador', () => {
    let wrapper
    // beforeEach() sirbe para tener una nueva instancia, se ejecuta antes de cada test
    beforeEach(() => {
        wrapper = shallow(<Contador/>)
    })

    test('Validando Snapshot', ()=>{
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('Validando funcionamiento de Botones', ()=>{
        // con simulate() podemos simular eventos
        wrapper.find('button').first().simulate('click')
        expect(wrapper.find('h3').text()).toBe("1")
    })
})