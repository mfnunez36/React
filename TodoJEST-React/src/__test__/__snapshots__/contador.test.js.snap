// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando Componente Contador Validando Snapshot 1`] = `
<div>
  <h3>
    0
  </h3>
  <button
    onClick={[Function]}
  >
    increment
  </button>
  <button
    onClick={[Function]}
  >
    decrement
  </button>
</div>
`;
